src/test/rbf_tests.cpp

```diff
@@ -0,0 +1,225 @@
+    // Additional fees must cover the replacement's vsize at incremental relay fee
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 1, 2, incremental_relay_feerate, unused_txid).has_value());
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 2, 2, incremental_relay_feerate, unused_txid) == std::nullopt);
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 2, 2, higher_relay_feerate, unused_txid).has_value());
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 4, 2, higher_relay_feerate, unused_txid) == std::nullopt);
+    BOOST_CHECK(PaysForRBF(low_fee, high_fee, 99999999, incremental_relay_feerate, unused_txid).has_value());
+
+    // Tests for GetEntriesForConflicts
+    CTxMemPool::setEntries all_parents{entry1, entry3, entry5, entry7, entry8};
+    CTxMemPool::setEntries all_children{entry2, entry4, entry6};
+    std::vector<CTransactionRef> parent_inputs({m_coinbase_txns[0], m_coinbase_txns[1], m_coinbase_txns[2],
+                                                m_coinbase_txns[3], m_coinbase_txns[4]});
+    CTransactionRef conflicts_with_parents = make_tx({50 * CENT}, parent_inputs);
+    CTxMemPool::setEntries all_conflicts;
+    BOOST_CHECK(GetEntriesForConflicts(/*tx=*/ *conflicts_with_parents.get(),
+                                       /*pool=*/ pool,
+                                       /*iters_conflicting=*/ all_parents,
+                                       /*all_conflicts=*/ all_conflicts) == std::nullopt);
+    BOOST_CHECK(all_conflicts == all_entries);
+    auto conflicts_size = all_conflicts.size();
+    all_conflicts.clear();
+
+    add_descendants(tx2, pool, 23);
+    BOOST_CHECK(GetEntriesForConflicts(*conflicts_with_parents.get(), pool, all_parents, all_conflicts) == std::nullopt);
+    conflicts_size += 23;
+    BOOST_CHECK_EQUAL(all_conflicts.size(), conflicts_size);
+    all_conflicts.clear();
+
+    add_descendants(tx4, pool, 23);
+    BOOST_CHECK(GetEntriesForConflicts(*conflicts_with_parents.get(), pool, all_parents, all_conflicts) == std::nullopt);
+    conflicts_size += 23;
+    BOOST_CHECK_EQUAL(all_conflicts.size(), conflicts_size);
+    all_conflicts.clear();
+
+    add_descendants(tx6, pool, 23);
+    BOOST_CHECK(GetEntriesForConflicts(*conflicts_with_parents.get(), pool, all_parents, all_conflicts) == std::nullopt);
+    conflicts_size += 23;
+    BOOST_CHECK_EQUAL(all_conflicts.size(), conflicts_size);
+    all_conflicts.clear();
+
+    add_descendants(tx7, pool, 23);
+    BOOST_CHECK(GetEntriesForConflicts(*conflicts_with_parents.get(), pool, all_parents, all_conflicts) == std::nullopt);
+    conflicts_size += 23;
+    BOOST_CHECK_EQUAL(all_conflicts.size(), conflicts_size);
+    BOOST_CHECK_EQUAL(all_conflicts.size(), 100);
+    all_conflicts.clear();
+
+    // Exceeds maximum number of conflicts.
+    add_descendants(tx8, pool, 1);
+    BOOST_CHECK(GetEntriesForConflicts(*conflicts_with_parents.get(), pool, all_parents, all_conflicts).has_value());
```

Comment:
Bikeshedding of the day. This function name could actually reflect the fact that BIP125 is enforced here for codebase clarity, something like `FetchConflictingEntriesAndCheckMaxReplacement` ? (can be in other commit/PR).

Reply:
Agree the naming could be changed to reflect that the function checks max replacements. But maybe for another PR :sweat_smile: